core/src/main/java/com/microsoft/applicationinsights/internal/jmx/JmxDataFetcher.java

```diff
@@ -72,14 +73,25 @@
                 Collection<Object> resultForAttribute = fetch(server, objects, attribute.attribute);
                 result.put(attribute.metricName, resultForAttribute);
             } catch (Exception e) {
-                logger.warn("Failed to fetch JMX object '{}' with attribute '{}': ", objectName, attribute.attribute);
-                throw e;
+                //Is it ok if we print out the exception text instead of throwing the exception so we can continue the loop?
+                //Is it ok to print out and log the exception once per metric so that we do not flood the logs?
+                if(!warningShown.contains(attribute)) {
+                    warningShown.add(attribute);
+                    System.out.println(e.toString());
+                    logger.error(e.toString());
+                    logger.warn("Failed to fetch JMX object '{}' with attribute '{}': ", objectName, attribute.attribute);
+                }
+
+                //throw e;
             }
         }
 
         return result;
     }
 
+    // objectName = "gc=*"
+    // attribute = "collectionTime"
+
     private static Collection<Object> fetch(MBeanServer server, Set<ObjectName> objects, String attributeName)
             throws AttributeNotFoundException, MBeanException, ReflectionException, InstanceNotFoundException {
```

Comment:
I like the idea of a comment here with example parameters, but it's a little unclear

Reply:
just checking you saw this (github is hiding it behind "2 hidden conversations" button)

Reply:
@trask Did not see this... this is a placeholder for the future work, I can remove it, not needed here